scripts/src/public/services/offers.ts

```diff
@@ -33,56 +35,77 @@ export interface OfferList {
 	paging: Paging;
 }
 
-function offerDbToApi(offer: db.Offer, content: db.OfferContent) {
-	content.content = offerContents.replaceTemplateVars(offer, content.content);
-	return {
+type OfferTranslations = {
+	title: string,
+	description: string,
+	orderTitle: string,
+	orderDescription: string,
+	content: any
+};
+
+function offerDbToApi(offer: db.Offer, content: db.OfferContent, offerTranslations: OfferTranslations) {
+	const offerData = {
 		id: offer.id,
-		title: offer.meta.title,
-		description: offer.meta.description,
+		title: offerTranslations.title || offer.meta.title,
+		description: offerTranslations.description || offer.meta.description,
 		image: offer.meta.image,
 		amount: offer.amount,
 		blockchain_data: offer.blockchainData,
 		offer_type: offer.type,
-		content: content.content,
+		content: offerTranslations.content || content.content,
 		content_type: content.contentType,
 	};
+	offerData.content = offerContents.replaceTemplateVars(offer, offerData.content);
+	return offerData;
 }
 
 /**
  * return the sublist of offers from this app that the user can complete
  */
-async function filterOffers(userId: string, app: Application | undefined, logger: LoggerInstance): Promise<Offer[]> {
+async function filterOffers(userId: string, app: Application | undefined, logger: LoggerInstance, acceptsLanguagesFunc?: any): Promise<Offer[]> {
 	// TODO: this should be a temp fix!
 	// the app should not be undefined as we used left join, figure it out
-	if (!app) {
-		return [];
-	}
-
-	if (app.offers.length === 0) {
+	if (!app || !app.offers.length) {
 		return [];
 	}
+	// The acceptsLanguagesFunc returns an array of all client accepted languages if no params are passed. If an array of languages is passed the when most suitable for the client will be returned.
 	const offerCounts = await Order.countAllByOffer(userId);
 	const contents = await offerContents.getAllContents();
-
+	let availableTranslations: OfferTranslation[] = [];
+	let language: string | null = null;
+	if (acceptsLanguagesFunc) {
+		availableTranslations = await OfferTranslation.createQueryBuilder("translations")
+			.where("translations.language IN (:languages)", { languages: acceptsLanguagesFunc() })
+			.getMany();
+		const availableLanguages = new Set(availableTranslations.map(translation => translation.language));
+		language = acceptsLanguagesFunc(Array.from(availableLanguages)); // get the most suitable language for the client
+	}
 	return (await Promise.all(
 		app.offers
 			.map(async offer => {
-				if ((offerCounts.get(offer.id) || 0) >= offer.cap.per_user) {
-					return null;
+					if ((offerCounts.get(offer.id) || 0) >= offer.cap.per_user) {
+						return null;
+					}
+					const content = contents.get(offer.id);
+					if (!content) {
+						return null;
+					}
+					let offerTranslations: OfferTranslations = {} as OfferTranslations;
```

Comment:
either
`let offerTranslations: OfferTranslations = {};`
or
`let offerTranslations = {} as OfferTranslations;`
no need for both